vpr/src/draw/search_bar.cpp

```diff
@@ -339,34 +420,135 @@ void warning_dialog_box(const char* message) {
 void search_type_changed(GtkComboBox* self, ezgl::application* app) {
     auto type = gtk_combo_box_text_get_active_text(GTK_COMBO_BOX_TEXT(self));
     GtkEntry* searchBar = GTK_ENTRY(app->get_object("TextInput"));
+    GtkEntryCompletion* completion = GTK_ENTRY_COMPLETION(app->get_object("Completion"));
+    GtkTreeModel* netNames = GTK_TREE_MODEL(app->get_object("NetNames"));
+    GtkTreeModel* blockNames = GTK_TREE_MODEL(app->get_object("BlockNames"));
     //Ensuring a valid type was selected
     if (!type) return;
     if (type[0] == '\0') return;
     std::string searchType(type);
-    //Setting active completion model to blockCompleter if type selected is block Name
+
+    /*
+     * If search type is name, connecting search bar to completion,
+     * and connecting completion to the appropriate model (blocks or nets)
+     * Additionally, visibility of key length setter is toggled by these changes
+     */
     if (searchType == "Block Name") {
-        GtkEntryCompletion* blockCompleter = GTK_ENTRY_COMPLETION(app->get_object("BlockNameCompleter"));
-        gtk_entry_set_completion(searchBar, blockCompleter);
-    } else { //If not, setting to null
+        gtk_entry_completion_set_model(completion, blockNames);
+    } else if (searchType == "Net Name") {
+        gtk_entry_completion_set_model(completion, netNames);
+    } else { //setting to null if option does not require auto-complete
+        gtk_entry_completion_set_model(completion, NULL);
         gtk_entry_set_completion(searchBar, nullptr);
     }
 }
 
 /**
- * @brief loads block names into gtk list store item used for completion
+ * @brief A non-default matching function. As opposed to simply searching for a prefix(default),
+ * searches string for presence of a substring. Ignores cases.
  * 
- * @param app ezgl application
+ * @param completer the GtkEntryCompletion being used
+ * @param key a normalized and case-folded key representing the text
+ * @param iter GtkTreeIter pointing at the current entry being compared
+ * @param user_data null
+ * @return gboolean 
```

Comment:
When is true vs. false returned?

Reply:
My guess of return value comment:
Returns true if the (single) text entry pointed at by GtkTreeIter contains key as a case-insensitive substring; false otherwise.